src/app/server/create-server.js

```diff
@@ -151,6 +151,12 @@ function createServer (app, BrowserWindow, ipcMain, opts) {
     })
   })
 
+  app.on('certificate-error', (event, webContents, url, error, certificate, callback) => {
```

Comment:
> you mentioned this practice here wasn't recommend and that you'll look for alternatives

True, but I dug deeper, and in our case:

  - We don't request external content anywhere in image-exporter, except for "dash-apps".
  - `app.on('certificate-error', ...` will ensure that any dash-app URLs (local or public IPs) are not blocked by imageserver because of certificates (or their absence).
  - `info.browserOpts.webPreferences = {nodeIntegration: false}` will ensure that the remote window used for rendering Dash Apps do not have access to the node process (renderer or main process), hence the dash-app content cannot do any malicious action..

> Moreover, I'm sure some users would prefer failing on http pages and other things that cause certificate errors, so I'm thinking we should make this optional by adding an API/CLI option.

That sounds good to me.. I can add it in this PR.. !!

Reply:
@tarzzz asked me for my thoughts on this so here they are...

> We don't request external content anywhere in image-exporter, except for "dash-apps".

Are you sure? Plots can embed external images for one thing. Also we've discussed adding this ability to dashboards.

Adding this option for Dash Apps seems reasonable to me provided:

1. It defaults to "off" in On-Prem 2.4.0.
2. The change is only made for Dash Apps.

I'm not sure from #10821 exactly what problem you're trying to solve, and I've asked some questions in that issue. Depending on the requirements, this alternative solution may work:
* Make the On-Prem certificates available to `image-exporter`.
* Document that mixed-content (http content in an https page) is not allowed. Browsers are removing this ability by default anyway.

However if it's less work to ignore cert errors that seems reasonable under the conditions I listed above.